core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/client/transactional/TransactionManager.java

```diff
@@ -2232,7 +2232,7 @@ public void doCommitDDL(final TransactionState transactionState) throws Unsucces
                         if (retry) 
                             retrySleep = retry(retrySleep);
                     }
-                } while (retry == true);
+                } while (retry && retryCount++ <= RETRY_ATTEMPTS);
```

Comment:
The code you have is fine. It could be made a bit simpler though perhaps by using just one variable, retryCount, and counting down instead of up. You could then simply test retryCount > 0. In code paths where you decide to not retry, just set retryCount = 0.